libs/language-server/src/lib/lsp/jayvee-scope-provider.ts

```diff
@@ -40,22 +52,103 @@ export class JayveeScopeProvider extends DefaultScopeProvider {
       return EMPTY_SCOPE;
     }
 
+    // TODO: add caching to avoid performance issues
+
     const importedUris = new Set<string>();
     this.gatherImports(jayveeModel, importedUris);
     this.gatherBuiltins(importedUris);
 
-    const importedElements = this.indexManager.allElements(
-      referenceType,
-      importedUris,
+    const importedDocuments = [...importedUris].map((importedUri) =>
+      this.langiumDocuments.getDocument(URI.parse(importedUri)),
     );
 
+    const importedElements: AstNodeDescription[] = [];
+    for (const importedDocument of importedDocuments) {
+      if (importedDocument === undefined) {
+        continue;
+      }
+
+      const publishedElements = this.getExportedElements(importedDocument);
+      importedElements.push(
+        ...publishedElements.map((e) =>
+          this.descriptions.createDescription(e, e.name),
+        ),
+      );
+    }
+
     return new MapScope(importedElements);
   }
 
   /**
-   * Add all builtins URIs to @param importedUris
+   * Gets all exported elements from a document.
+   * This logic cannot reside in a {@link ScopeComputationProvider} but should be handled here:
+   * https://github.com/eclipse-langium/langium/discussions/1508#discussioncomment-9524544
+   */
+  protected getExportedElements(
+    document: LangiumDocument,
+  ): ExportableElement[] {
+    const model = document.parseResult.value as JayveeModel;
+    const exportedElements: ExportableElement[] = [];
+
+    for (const node of AstUtils.streamAllContents(model)) {
+      if (isExportableElement(node) && node.isPublished) {
+        exportedElements.push(node);
+      }
+
+      if (isExportDefinition(node)) {
+        const originalDefinition = this.followExportDefinitionChain(node);
+        if (originalDefinition !== undefined) {
+          exportedElements.push(originalDefinition);
+        }
+      }
+    }
+    return exportedElements;
+  }
+
+  /**
+   * Follow an export statement to its original definition.
+   */
+  protected followExportDefinitionChain(
```

Comment:
If we should ever allow `publish * from '{path}'`, we'd have to use recursion here I think to follow the chain.. Right now, the exported element is directly referenced.